lib/src/Uncertainty/Distribution/openturns/UniformOrderStatistics.hxx

```diff
@@ -0,0 +1,164 @@
+
+  /** Compute the entropy of the distribution */
+  Scalar computeEntropy() const override;
+
+  /** Get the standard deviation of the distribution */
+  Point getStandardDeviation() const override;
+
+  /** Get the skewness of the distribution */
+  Point getSkewness() const override;
+
+  /** Get the kurtosis of the distribution */
+  Point getKurtosis() const override;
+
+  /** Parameters value accessors */
+  Point getParameter() const override;
+
+  /** Parameters description accessor */
+  Description getParameterDescription() const override;
+
+  /* Interface specific to UniformOrderStatistics */
+
+  /** Compute the PDF of Xi | X1, ..., Xi-1. x = Xi, y = (X1,...,Xi-1) */
+  using ContinuousDistribution::computeConditionalPDF;
+  Scalar computeConditionalPDF(const Scalar x, const Point & y) const override;
+  Point computeSequentialConditionalPDF(const Point & x) const override;
+
+  /** Compute the CDF of Xi | X1, ..., Xi-1. x = Xi, y = (X1,...,Xi-1) */
+  using ContinuousDistribution::computeConditionalCDF;
+  Scalar computeConditionalCDF(const Scalar x, const Point & y) const override;
+  Point computeSequentialConditionalCDF(const Point & x) const override;
+
+  /** Compute the quantile of Xi | X1, ..., Xi-1, i.e. x such that CDF(x|y) = q with x = Xi, y = (X1,...,Xi-1) */
+  using ContinuousDistribution::computeConditionalQuantile;
+  Scalar computeConditionalQuantile(const Scalar q, const Point & y) const override;
+  Point computeSequentialConditionalQuantile(const Point & q) const override;
+
+  /** Get the i-th marginal distribution */
+  Distribution getMarginal(const UnsignedInteger i) const override;
+
+  /** Get the distribution of the marginal distribution corresponding to indices dimensions */
+  Distribution getMarginal(const Indices & indices) const override;
+
+  /** Tell if the distribution has independent marginals */
+  Bool hasIndependentCopula() const override;
+
+  /** Tell if the distribution has an elliptical copula */
+  Bool hasEllipticalCopula() const override;
+
+  /** Get the Spearman correlation of the distribution */
+  CorrelationMatrix getSpearmanCorrelation() const override;
+
+  /** Get the Kendall concordance of the distribution */
+  CorrelationMatrix getKendallTau() const override;
+
+  /** Method save() stores the object through the StorageManager */
+  void save(Advocate & adv) const override;
+
+  /** Method load() reloads the object from the StorageManager */
+  void load(Advocate & adv) override;
+
+
+protected:
+
+private:
+
+  /** Compute the mean of the distribution */
+  void computeMean() const override;
+
+  /** Compute the covariance of the distribution */
+  void computeCovariance() const override;
+
+  /** Get the quantile of the distribution */
+  Scalar computeScalarQuantile(const Scalar prob, const Bool tail = false) const override;
+
+  /** Compute the numerical range of the distribution given the parameters values */
+  void computeRange() override;
```

Comment:
Is there a particular reason why these methods are private rather than protected?